src/PointwiseFunctions/AnalyticSolutions/Burgers/Step.cpp

```diff
@@ -0,0 +1,77 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/Burgers/Step.hpp"
+
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/Math.hpp"  // IWYU pragma: keep
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace Burgers {
+namespace Solutions {
+
+Step::Step(const double left_value, const double right_value,
+           const double initial_shock_position) noexcept
+    : left_value_(left_value),
+      right_value_(right_value),
+      initial_shock_position_(initial_shock_position) {
+  ASSERT(left_value > right_value,
+         "Shock solution expects left_value > right_value");
```

Comment:
PARSE_ERROR (because checking input file values)

Reply:
Can you clarify what change you have in mind? It looks like PARSE_ERROR expects an OptionContext; how do I get this from the constructor?

Reply:
Add a `const OptionContext& context = {}` argument as the last constructor argument.  The option parser will detect this and pass one.